### PR TITLE
Add no-archive format for fast environment clone

### PR DESCRIPTION
**Pull request by ayurchuk**
Existing formats do not provide a straightforward way to clone the environment without making an intermediate archive. Even with disabled compression, they impact execution time and use extra disk space. The no-archive is a new format that addresses both aspects. It is 4x faster for local environment cloning on basic scenarios (i.e. github.com/idamlaj/dist-demo) and uses no extra disk space.

### CODE REVIEW

**Comment by ayurchuk** [comment]
@conda-bot

**Comment by xhochy** [comment]
@ayurchuk See the failing unit tests.

**Comment by ayurchuk** [comment]
@xhochy thanks, should be addressed now. Side question, what is the best way to trigger tests on the PR?

**Comment by xhochy** [comment]
As this is your first PR to the repository, a maintainer needs to approve the run first on each new commit.

**Comment by ayurchuk** [comment]
@xhochy I'm wondering if there is any page on how to run test suite locally

File: conda_pack/formats.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def _parse_n_threads(n_threads=1):
     return n_threads
 
 
-def archive(fileobj, path, arcroot, format, compress_level=4, zip_symlinks=False,
+def archive(output, fileobj, path, arcroot, format, compress_level=4, zip_symlinks=False,
             zip_64=True, n_threads=1, verbose=False):
```
Comment (xhochy) [inline conda_pack/formats.py:32]: Can you please make this change non-breaking?
```suggestion
def archive(fileobj, path, arcroot, format, compress_level=4, zip_symlinks=False,
            zip_64=True, n_threads=1, verbose=False, output=None):
```

**Pull request merged**